File: CMFexacto.py
from funcionesAux import dameCliques
from funcionesAux import frontera

#CMFgoloso: Busca todas las cliques del grafo, compara sus fronteras y devuelve la mayor.
def CMFexacto(grafo, n):
	#Grafo = Matriz, n = #Nodos.

	cliques = dameCliques(grafo, range(n))

	fronteras = []
	for clique in cliques:

		fronteras.append(frontera(grafo, n, clique))

	return max(fronteras)


n = 4
G = [
[False, True, False, True],
[True, False, True, True],
[False, True, False, False],
[True, True, False, False]
]


print CMFexacto(G, n)

File: CMFexacto.java
import java.util.*;

public class CMFexacto
{

    public static List<Integer> CMFexacto(boolean[][] grafo, int n)
    {	

    	List<Integer> nodos = new ArrayList<Integer>();
    	for(int i=0;i<n;i++)
        {
        	nodos.add(i);
        }

		List<List<Integer>> cliques = funcionesAux.dameCliques(grafo, nodos);

		int fronteraMax = 0;
		List<Integer> cliqueMax = new ArrayList<Integer>();
    	
    	for(List<Integer> clique : cliques)
        {	
        	int frontera = funcionesAux.frontera(grafo, n, clique);
        	if(fronteraMax < frontera)
        	{
        		fronteraMax = frontera;
        		cliqueMax = clique;

        	}
        	
        }

		return cliqueMax;
    }


    public static void main(String[] args)
    {	
    	Scanner s = new Scanner(System.in);

        int n = s.nextInt();
        int m = s.nextInt();

        boolean[][] grafo = EntradaSalida.leerGrafoEntrada(n, m);

        List<Integer> clique = CMFexacto(grafo, n);
        int sizeClique = clique.size();
        int frontera = funcionesAux.frontera(grafo, n, clique);

        System.out.print(frontera+" "+sizeClique+" ");
        for(int nodo : clique)
        {
        	System.out.print(nodo+" ");
        }

    }

}